artemis-protocols/artemis-openwire-protocol/src/main/java/org/apache/activemq/artemis/core/protocol/openwire/amq/AMQSession.java

```diff
@@ -133,7 +133,8 @@ public void initialize() {
       // now
 
       try {
-         coreSession = server.createSession(name, username, password, minLargeMessageSize, connection, true, false, false, false, null, this, true, connection.getOperationContext(), protocolManager.getPrefixes(), protocolManager.getSecurityDomain());
+         final String validatedUser = server.validateUser(username, password, connection, protocolManager.getSecurityDomain());
```

Comment:
if OpenWireProtocolManager.validateUser could set a new field as OpenWireConnection.validatedUser the AMQSession.initialize shouldn't execute server.validateUser again

Reply:
that is wise. thanks